fecgov/fec-cms
# Feature/3517 presidential map 3

## Summary

- Resolves #3517 

_This is V3 of the presidential map changes. This branch is deployed to the feature space for testing._

## Impacted areas of the application

List general components of the application that this PR will affect:

-  The Presidential Campaign Finance Map
- Where Contributions Come From

## Screenshots

![image](https://user-images.githubusercontent.com/26720877/76560836-66b1c180-6478-11ea-9840-c64e12e8a866.png)

## Related PRs

There will be one for the next pres map branch (4)

## How to test

- Like always, pull the branch, `npm i`, `npm run build`, `./manage.py runserver`
- Make sure [WCCF](http://127.0.0.1:8000/data/raising-bythenumbers) works as it should
  - Change candidates and elections
  - Verify that data is generally correct (and changes from this PR would affect the displayed map, not the data behind it)
  - Check that states aren't clickable
  - but rollovers still work
  - and circles don't appear, ever
- Put the [pres map](http://127.0.0.1:8000/data/candidates/president/presidential-map/) through its paces
  - Change candidates
  - Change years
    - Also, change years when a candidate is selected who ran in both years
    - And when the selected candidate did _not_ run in both years
  - Click different states
    - in gradient view
    - in bubbles view
    - while zoomed out
    - while zoomed in
  - Check the accordions
  - Check the downloads
- All of the above on desktop
- All of the above on medium widths
- All of the above on small widths

## Known issues

- The items in the MoSCoW
- Some appearance styles need polish (but please list them!)

## Next steps
- need to add error handling
- should visually let the user know that data loads are still ongoing (and limit their interaction)
- analytics
- the MoSCow
- remote table header for candidates list
- remove comments
- remove full-doc eslint exemptions
- add documentation

____
